test/test_helper.rb

```diff
@@ -86,8 +87,22 @@ def after_teardown
   end
 end
 
+module ExtensibleFixtureSupport
+  extend ActiveSupport::Concern
+
+  included do
+    self.fixture_path = "#{Rails.root}/tmp/combined_fixtures/"
+
+    directories = ["#{Rails.root}/test/fixtures"] + Foreman::Plugin.all.map{ |p| p.fixture_directory }
+    helper = FixtureHelper.new(directories.compact)
+    helper.combine(self.fixture_path)
+    fixtures(:all)
+  end
```

Comment:
We need to take great attention to this code, fixtures gets loaded for *every test* and we already have slow test suite. We must be sure there are no extra copies of yaml files somewhere. The code looks correct, but I still think this is unncecessary complex.

We have this in `test/active_support_test_case_helper.rb` file:

```ruby
 class ActiveSupport::TestCase
    # Setup all fixtures in test/fixtures/*.(yml|csv) for all tests in alphabetical order.
    # Note: You'll currently still have to declare fixtures explicitly in integration tests
    # -- they do not yet inherit this setting
    fixtures :all
...
```

I think the whole patch is based on what was in Katello, I believe we can do better than this and simply setup fixtures in a correct way without need of copying things over. I don't see any technical limitation of the ActiveSupport stack.

Reply:
Its my understanding that ths code only gets run at class load time, but i will dig further to see how we could specify all the files themselves.

Reply:
Sure, but as I read it, it only prepares an array of yaml files. In the worst case we should be able to call this twice. But it it turns out it's too difficult, we can go the way you propose, I just think it is worth taking a look.

Reply:
yes, when you run 'rake test', this code is only executed once

Reply:
What if Rails 6 loads fixtures on every test or something, we won't notice and tests will be copying tree hierarchy for every single test. Can't think how to create a test for that honestly, maybe some safety check in the loading code. Any ideas? I just want to have the safest possible hack :-)

Reply:
I guess we could use a class variable to store whether they had been loaded or not and not run twice.